# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/TlsHandshaker.cpp
# Commit: 373bc48d5325
# Full Hash: 373bc48d5325e17cb67a11035d4542d221023594
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2023-09-19 09:37:28
# Regressor Bug: 1784261
# File Overlap Count: 1
# Description:
#   Bug 1784261 - Add telemetry to understand the reason of connection closure, r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D188132
# ==============================================================================

diff -r f2fee3884997 -r 373bc48d5325 netwerk/protocol/http/TlsHandshaker.cpp
--- a/netwerk/protocol/http/TlsHandshaker.cpp	Mon Sep 18 14:28:30 2023 +0000
+++ b/netwerk/protocol/http/TlsHandshaker.cpp	Mon Sep 18 14:44:18 2023 +0000
@@ -258,6 +258,7 @@
          "early selected alpn not available",
          mOwner.get()));
   } else {
+    mOwner->ChangeConnectionState(ConnectionState::ZERORTT);
     LOG1(
         ("TlsHandshaker::Check0RttEnabled %p -"
          "early selected alpn: %s",